### PR TITLE
LPD-18582 - In Workflow if the script management setting is true show warning alert and disable the update

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://liferay.atlassian.net/browse/LPD-18582

### CODE REVIEW

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js [hunk 1]
```diff
@@ -238,6 +257,16 @@ export default function UpperToolbar({
 	};
 
 	const saveDefinition = () => {
+		if (
+			Liferay.FeatureFlags['LPD-11179'] &&
+			!allowScriptContentBeExecutedOrIncluded &&
+			detectGroovyScript(elements, setHasGroovyScript)
+		) {
+			setShowGroovyScriptWarningModal(true);
+
```
[the remark this answers is not in the document]
Comment (cgmonte) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:266]: But remember to only set to false inside the `.then()`, if `response.ok`. You can store the result of:

```
Liferay.FeatureFlags['LPD-11179'] &&
!allowScriptContentBeExecutedOrIncluded &&
detectGroovyScript(elements, setHasGroovyScript)
```

inside a variable and use both on the `if` above and inside `then`.

**Review by cgmonte — CHANGES_REQUESTED** [review]
Hi @danvb112, great work as always. I have made some suggestions to avoid code repetition, but I am not sure if they are worth. About the bug we found, I made a comment where I think you should add the fix, but again, I am not sure that is the right spot.

**Comment by danvb112** [comment]
ci:test:sf

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/shared-components/ScriptInput.js [hunk 2]
```diff
@@ -26,11 +28,30 @@ const ScriptInput = ({
...
+	const {hadGroovyScriptBefore} = useContext(DiagramBuilderContext);
+	const {allowScriptContentBeExecutedOrIncluded} = useContext(
+		DefinitionBuilderContext
+	);
 	const [script, setScript] = useState(inputValue);
 	const [scriptLanguage, setScriptLanguage] = useState(
 		defaultScriptLanguage || DEFAULT_LANGUAGE
 	);
 
+	const getScriptLanguageOptions = () => {
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/shared-components/ScriptInput.js]: Hey @cgmonte, this is a great suggestion! I've added a new commit extracting this logic to a util and reusing the code :smile:

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/shared-components/ScriptInput.js [hunk 3]
```diff
@@ -46,7 +67,7 @@ const ScriptInput = ({
 				}}
 				onClickCapture={() => handleClickCapture(scriptLanguage)}
 			>
-				{scriptLanguageOptions.map((item) => (
+				{getScriptLanguageOptions().map((item) => (
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/shared-components/ScriptInput.js]: Done!

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/shared/components/toolbar/UpperToolbar.js:274]: Hey @cgmonte, your suggestion is right, I've added a commit to fix this bug using your suggestion, I had to put this logic in the `publishDefinition` and in the `saveDefinitnion` functions. Thanks for notice this! :smile:

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:workflow

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by danvb112** [comment]
ci:forward:force

**Pull request closed without merging**